src/v1/apis/oauth/services/google.oauth.service.ts

```diff
@@ -0,0 +1,116 @@
+import { OAuthRepositoryInterface } from '../../../storage/database/interfaces/oauth.repository.interface.js';
+import {
+  HandleCallbackInputType,
+  HandleCallbackResponseType,
+  OAuthService,
+} from './oauth.service.js';
+import TokenService from '../../auth/services/token.service.js';
+import { UnAuthorizedException } from '../../../common/exceptions/core.error.js';
+import OAuthUserService from './oauth.user.service.js';
+import OAuthCacheRepository from '../../../storage/cache/oauth.cache.repository.js';
+import {
+  oauthProviderSchema,
+  OAuthUserInfoSchema,
+  OAuthUserInfoType,
+} from '../schema/oauth.schema.js';
+import { OAuthTokenResponseType } from '../schema/handle-callback.schema.js';
+import * as crypto from 'node:crypto';
+import GoogleOauthClient from './googleOauthClient.js';
+import { OAuth2Client } from 'google-auth-library';
+
+export default class GoogleOauthService implements OAuthService {
+  constructor(
+    private readonly oauthRepository: OAuthRepositoryInterface,
+    private readonly oauthCacheRepository: OAuthCacheRepository,
+    private readonly tokenService: TokenService,
+    private readonly oauthUserService: OAuthUserService,
+    private readonly googleOauthClient: GoogleOauthClient,
+  ) {}
+
+  readonly provider = oauthProviderSchema.enum.GOOGLE;
```

Comment:
`provider`는 클래스의 설정 값(상수)에 가까우니, constructor 위에 선언해 주세요.

```suggestion
export default class GoogleOauthService implements OAuthService {
  readonly provider = oauthProviderSchema.enum.GOOGLE;
  
  constructor(
    private readonly oauthRepository: OAuthRepositoryInterface,
    private readonly oauthCacheRepository: OAuthCacheRepository,
    private readonly tokenService: TokenService,
    private readonly oauthUserService: OAuthUserService,
    private readonly googleOauthClient: GoogleOauthClient,
  ) {}

```